### DISCUSSION TITLE
BelongsTo subfields? How?

### DISCUSSION THREAD

**Discussion by kde99** [Laravel-Backpack/community-forum#664] [Q&A (Help)]
Hello everyone!

I have my two models called Partners(id, name, mail_address_id, shipping_address_id) and Addresses(id, country, address). According to the documentation this should be a belongsTo relationship (since partners has they address_id field).

My issue is that with belongsTo like this Backpack creates the select from a list type of field, but what I want is the subfields type. When I defined it by hand (with belongsTo) it is ignored and the select type shows still. I tried to define it with hasOne (I know it's not the right relationship BUT partner <-> address is 1-1! not n-1), this partially works: the subfields show up I can save (it saves the new address), but neither of the Partner table's fields get updated with the new address id.

I have also thought about the inline create operation, but I want the subfields to show up in case we have to change the addresses manually.

My question is: how could I make subfields work with this arrangement without having to create a whole new table for these address types?

**Comment by pxpm** [comment]
Hello @kde99 

For `BelongsTo` we have the [Inline Create Operation](https://backpackforlaravel.com/docs/6.x/crud-fields#create-related-entries-in-a-modal-using-the-inlinecreate-operati)

Does it work for you ? 

Cheers

**Reply by kde99** [reply]
Thank you for your quick answer. 

The inline create would be perfect, but from the demo it seems like there is no edit functionality to it, so if the address changes (meaning no need for the old one) we need to add a new one then delete the old somehow. Also I would like to keep this relationship 1-1 since that would make deletion of the partner model easier (no need to prune useless addresses).

Also with the select field showing you could see all the addresses but our current (old system) has around 1500 partners in it with almost 0 need to be able to select an address more than once. This is why I was thinking about asking if there was a way with this schema to make it work using subfields, but I guess we will have to roll with something different for these ones like [this hasMany example](https://backpackforlaravel.com/docs/6.x/crud-fields#manage-related-entries-in-the-same-form-create-update-delete) with an address type selection box.
